daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/PackedBinaryTraits.scala

```diff
@@ -70,11 +71,16 @@ abstract class PackedBinaryDecimalBaseParser(
 
 abstract class PackedBinaryIntegerBaseParser(
   override val context: ElementRuntimeData,
-  signed: Boolean = false
 ) extends PrimParser
   with PackedBinaryConversion {
   override lazy val runtimeDependencies = Vector()
 
+  val signed = {
+    context.optPrimType.get match {
+      case n: NodeInfo.PrimType.PrimNumeric => n.isSigned
+      case _ => false
+    }
+  }
   protected def getBitLength(s: ParseOrUnparseState): Int
 
   def parse(start: PState): Unit = {
```

Comment:
Just below we have this line:
```scala
if (nBits == 0) return // zero length is used for outputValueCalc often.
```
The purpose of this bug was to make it so zero length numbers is an error. I don't understand why outputValueCalc would have zero length, but based on the bug it sounds like that shouldn't be allowed any more. Should that change to an unparse error?

Reply:
Above this we have PackedDecimal that also returns if nBits is 0, should we also make that a PE? Also PackedDecimal doesn't pass along decimalSigned to the base, is it logical to do something like the below? If so, they I think we need to update the PackedDecimal classes  to pass along decimalSigned

```xml
          <xs:element name="num" type="xs:decimal"  dfdl:representation="binary" dfdl:binaryNumberRep="ibm4690Packed" dfdl:decimalSigned="yes" dfdl:binaryDecimalVirtualPoint="0"/>
```

Reply:
Yeah, I think you need to know if the decimal representation is expected to be signed or not. The error case of dfdl:decimalSigned="no" but the packed representation has a negative sign needs to be detected somehow. 
One could choose different parse primitives based on dfdl:decimalSigned yes/no or pass a boolean isSigned to a common primitive.